plugins/woocommerce-admin/client/products/fields/attribute-control/attribute-control.tsx

```diff
@@ -74,79 +94,89 @@ export const AttributeControl: React.FC< AttributeControlProps > = ( {
+						getAttributeId( attr ) !== getAttributeId( attribute )
 				)
 			);
-		} else {
-			recordEvent( 'product_remove_attribute_confirmation_cancel_click' );
+			onRemove( attribute );
+			return;
 		}
+		onRemoveCancel( attribute );
+	};
+
+	const openNewModal = () => {
+		setIsNewModalVisible( true );
+		onNewModalOpen();
+	};
+
+	const closeNewModal = () => {
+		setIsNewModalVisible( false );
+		onNewModalClose();
+	};
+
+	const openEditModal = ( attribute: ProductAttribute ) => {
+		setCurrentAttributeId( getAttributeId( attribute ) );
+		onEditModalOpen( attribute );
+	};
+
+	const closeEditModal = ( attribute: ProductAttribute ) => {
+		setCurrentAttributeId( null );
+		onEditModalClose( attribute );
 	};
 
-	const onAddNewAttributes = (
-		newAttributes: EnhancedProductAttribute[]
-	) => {
+	const handleAdd = ( newAttributes: EnhancedProductAttribute[] ) => {
 		handleChange( [
-			...( value || [] ),
-			...newAttributes
-				.filter(
-					( newAttr ) =>
-						! ( value || [] ).find( ( attr ) =>
-							newAttr.id === 0
-								? newAttr.name === attr.name // check name if custom attribute = id === 0.
-								: attr.id === newAttr.id
-						)
-				)
-				.map( ( newAttr, index ) => {
-					return {
-						...newAttributeProps,
-						...newAttr,
-						position: ( value || [] ).length + index,
-					};
-				} ),
+			...value,
+			...newAttributes.filter(
+				( newAttr ) =>
+					! value.find(
+						( attr ) =>
+							getAttributeId( newAttr ) === getAttributeId( attr )
+					)
+			),
 		] );
-		recordEvent( 'product_add_attributes_modal_add_button_click' );
-		setShowAddAttributeModal( false );
+		onAdd( newAttributes );
+		closeNewModal();
+	};
+
+	const handleEdit = ( updatedAttribute: ProductAttribute ) => {
+		const updatedAttributes = value.map( ( attr ) => {
+			if (
+				getAttributeId( attr ) === getAttributeId( updatedAttribute )
+			) {
+				return updatedAttribute;
+			}
+
+			return attr;
+		} );
+
+		onEdit( updatedAttribute );
+		handleChange( updatedAttributes );
+		closeEditModal( updatedAttribute );
 	};
 
 	if ( ! value.length ) {
 		return (
 			<>
 				<AttributeEmptyState
-					addNewLabel={
-						isOnlyForVariations
-							? __( 'Add options', 'woocommerce' )
-							: undefined
-					}
-					onNewClick={ () => {
-						recordEvent(
-							'product_add_first_attribute_button_click'
-						);
-						setShowAddAttributeModal( true );
-					} }
-					subtitle={
-						isOnlyForVariations
-							? __( 'No options yet', 'woocommerce' )
-							: undefined
-					}
+					addNewLabel={ text.newAttributeModalTitle }
+					onNewClick={ () => openNewModal() }
+					subtitle={ text.emptyStateSubtitle }
 				/>
-				{ showAddAttributeModal && (
-					<AddAttributeModal
+				{ isNewModalVisible && (
+					<NewAttributeModal
 						onCancel={ () => {
-							recordEvent( CANCEL_BUTTON_EVENT_NAME );
-							setShowAddAttributeModal( false );
+							closeNewModal();
+							onNewModalCancel();
```

Comment:
I think we should check that this is a function before trying to call it (see comment above).